src/main/kotlin/me/dkim19375/dkimgradle/util/CommonScripts.kt

```diff
@@ -26,25 +26,61 @@ package me.dkim19375.dkimgradle.util
 
 import me.dkim19375.dkimgradle.annotation.API
 import me.dkim19375.dkimgradle.delegate.TaskRegisterDelegate
+import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.Task
-import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.compile.JavaCompile
 import org.gradle.kotlin.dsl.named
+import org.gradle.language.jvm.tasks.ProcessResources
 import java.io.File
 
+/**
+ * Sets up the project with the specified [group] and [version]
+ *
+ * Adds the text encoding, replacements, and build shadow task (if the Shadow plugin is applied)
+ */
+fun Project.setup(group: String, version: String) {
+    this.group = group
+    this.version = version
+    // Tasks
+    addTextEncodingTask()
+    addReplacementsTask()
```

Comment:
Can you remove addReplacementsTask() from setup() as (by default) it's only used for Minecraft, and there's a separate Minecraft setup method?

Reply:
I'll probably add like "setJavaTargetVersion" and "setKotlinTargetVersion" later on as well to this, so it will have more than just setting the group and version